src/Simplex/Chat/Types.hs

```diff
@@ -226,10 +229,68 @@ defaultChatSettings = ChatSettings {enableNtfs = True}
 pattern DisableNtfs :: ChatSettings
 pattern DisableNtfs = ChatSettings {enableNtfs = False}
 
+data ChatPreferences = ChatPreferences
+  { voice :: Maybe Preference
+  -- image :: Maybe Preference,
+  -- file :: Maybe Preference,
+  -- delete :: Maybe Preference,
+  -- acceptDelete :: Maybe Preference,
+  -- edit :: Maybe Preference,
+  -- receipts :: Maybe Preference
+  }
+  deriving (Eq, Show, Generic, FromJSON)
+
+pattern DefaultChatPrefs :: ChatPreferences
+pattern DefaultChatPrefs = ChatPreferences {voice = Nothing}
+
+instance ToJSON ChatPreferences where
+  toJSON = J.genericToJSON J.defaultOptions {J.omitNothingFields = True}
+  toEncoding = J.genericToEncoding J.defaultOptions {J.omitNothingFields = True}
+
+instance ToField ChatPreferences where
+  toField = toField . safeDecodeUtf8 . LB.toStrict . J.encode
+
+instance FromField ChatPreferences where
+  fromField = fromTextField_ $ J.decode . LB.fromStrict . encodeUtf8
+
+data Preference = Preference
+  {enable :: PrefSwitch}
+  deriving (Eq, Show, Generic, FromJSON)
+
+instance ToJSON Preference where
+  toJSON = J.genericToJSON J.defaultOptions {J.omitNothingFields = True}
+  toEncoding = J.genericToEncoding J.defaultOptions {J.omitNothingFields = True}
+
+data PrefSwitch = PSOn | PSOff -- for example it can be extended to include PSMutual, that is only enabled if it's enabled by another party
+  deriving (Eq, Show, Generic)
+
+instance FromField PrefSwitch where fromField = fromBlobField_ strDecode
+
+instance ToField PrefSwitch where toField = toField . strEncode
+
+instance StrEncoding PrefSwitch where
+  strEncode = \case
+    PSOn -> "on"
+    PSOff -> "off"
+  strDecode = \case
+    "on" -> Right PSOn
+    "off" -> Right PSOff
+    r -> Left $ "bad PrefSwitch " <> B.unpack r
+  strP = strDecode <$?> A.takeByteString
+
+instance FromJSON PrefSwitch where
+  parseJSON = strParseJSON "PrefSwitch"
+
+instance ToJSON PrefSwitch where
+  toJSON = strToJSON
+  toEncoding = strToJEncoding
+
 data Profile = Profile
   { displayName :: ContactName,
     fullName :: Text,
-    image :: Maybe ImageData
+    image :: Maybe ImageData,
+    contactPreferences :: Maybe ChatPreferences,
+    userPreferences :: Maybe ChatPreferences
```

Comment:
this is incorrect, as we send Profile as a part of messages, so it should only include one of the following:
1. the full set of user preferences (if it's to be sent)
2. preference overrides per contact (if it's saved to the database)
3. the preferences received from the contact
 
In all cases it should be in one property `preferences`
```suggestion
    preferences :: Maybe ChatPreferences
```

Reply:
the other property should most likely be in Contact